### PR TITLE
[CI] [GHA] Introduce GHA Windows Conditional Compilation Pipeline

### PR DESCRIPTION
**Pull request by akashchi**
### Details:
This PR introduces the Windows Conditional Compilation pipeline in the GitHub Actions environment. It is mirroring the [Azure Windows Conditional Compilation pipeline](https://github.com/openvinotoolkit/openvino/blob/master/.ci/azure/windows_conditional_compilation.yml).

### Tickets:
 - 117541

### CODE REVIEW

**Review by mryzhov — APPROVED** [review]

File: .github/workflows/windows_conditional_compilation.yml [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+          python3 -m pip install -r ${{ env.OPENVINO_REPO }}/src/frontends/paddle/tests/requirements.txt
+
+      - name: Install MO dependencies
+        run: |
+          python3 -m pip install -r ${{ env.OPENVINO_REPO }}/tools/mo/requirements_mxnet.txt `
+          -r ${{ env.OPENVINO_REPO }}/tools/mo/requirements_caffe.txt `
+          -r ${{ env.OPENVINO_REPO }}/tools/mo/requirements_kaldi.txt `
+          -r ${{ env.OPENVINO_REPO }}/tools/mo/requirements_onnx.txt `
+          -r ${{ env.OPENVINO_REPO }}/tools/mo/requirements_tf2.txt `
+          -r ${{ env.OPENVINO_REPO }}/tools/mo/requirements_dev.txt
```
Comment (ilya-lavrenov) [inline .github/workflows/windows_conditional_compilation.yml:108]: I suppose we don't need all these dependencies for conditional compilation.

File: .github/workflows/windows_conditional_compilation.yml [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+#  push:
+#    paths-ignore:
+#      - '**/docs/**'
+#      - 'docs/**'
+#      - '**/**.md'
+#      - '**.md'
+#      - '**/layer_tests_summary/**'
+#      - '**/conformance/**'
+#    branches:
+#      - master
```
Comment (ilya-lavrenov) [inline .github/workflows/windows_conditional_compilation.yml:24]: remove?

File: .github/workflows/windows_conditional_compilation.yml [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        shell: pwsh
+    runs-on: windows-latest-8-cores
+    steps:
+      - name: Clone OpenVINO
+        uses: actions/checkout@v3
+        with:
+          path: 'openvino'
+          submodules: 'recursive'
+
+      - name: Clone testdata for C API tests
```
Comment (ilya-lavrenov) [inline .github/workflows/windows_conditional_compilation.yml:65]: we don't have C API tests in CC build

File: .github/workflows/windows_conditional_compilation.yml [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+      #
+      # Dependencies
+      #
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install python dependencies
```
Comment (ilya-lavrenov) [inline .github/workflows/windows_conditional_compilation.yml:81]: python dependencies are not required for CC build

File: .github/workflows/windows_conditional_compilation.yml [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+          -r ${{ env.OPENVINO_REPO }}/tools/mo/requirements_caffe.txt `
+          -r ${{ env.OPENVINO_REPO }}/tools/mo/requirements_kaldi.txt `
+          -r ${{ env.OPENVINO_REPO }}/tools/mo/requirements_onnx.txt `
+          -r ${{ env.OPENVINO_REPO }}/tools/mo/requirements_tf2.txt `
+          -r ${{ env.OPENVINO_REPO }}/tools/mo/requirements_dev.txt
+
+      - name: Install build dependencies
+        run: |
+          choco install --no-progress ninja
+          choco install --no-progress shellcheck
```
Comment (ilya-lavrenov) [inline .github/workflows/windows_conditional_compilation.yml:113]: shellcheck is not required for CC

File: .github/workflows/windows_conditional_compilation.yml [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+      - name: Get tools versions
+        run: |
+          python3 --version
+          cmake --version
```
Comment (ilya-lavrenov) [inline .github/workflows/windows_conditional_compilation.yml:118]: drop

File: .github/workflows/windows_conditional_compilation.yml [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+      - name: List bin files
+        shell: cmd
+        run: dir ${{ env.OPENVINO_REPO }}\bin\ /s
+
+      - name: Code usage analysis
+        shell: cmd
+        working-directory: ${{ env.OPENVINO_REPO }}
+        run: |
+          set path=%path%;${{ env.OPENVINO_REPO }}\temp\tbb\bin
+          call "${{ env.VCVARSPATH }}" && python thirdparty\itt_collector\runtool\sea_runtool.py --bindir ${{ env.OPENVINO_REPO }}\bin\intel64\${{ env.BUILD_TYPE }} -o ${{ env.BUILD_DIR }}\itt_stat ! ${{ env.OPENVINO_REPO }}\bin\intel64\${{ env.BUILD_TYPE }}\benchmark_app.exe -niter 1 -nireq 1 -m ${{ env.MODELS_PATH }}\models\test_model\test_model_fp32.xml -d CPU
```
Comment (ilya-lavrenov) [inline .github/workflows/windows_conditional_compilation.yml:169]: similar comment about test_model

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**